asa_parser/src/asa_parse.py

```diff
@@ -200,9 +200,55 @@ def show_memory_region(self):
                 memory_region.append(mem)
         return json.dumps(memory_region)
 
+    @property
     def show_cpu_detailed(self):
-        """Parser for show cpu detailed"""
-        return json.dumps({'text': self.get_show_section('cpu detailed')})
+        """
+        Parser for show cpu detailed
+        The sections of show cpu detailed are parsed according to the break down of per core data path,
+        current control point elapsed,cpu utilization for external processes, and total cpu utilization
+        """
+        cpu_detailed = []
+        #Dictionary to hold core data
+        cores_data = {}
+
+        breakdown = {}
+        total_cpu_utilization = {}
+        external_cpu_utilization = {}
+        current_control_point = {}
+
+        prev_line = ""
+        temp_string = ""
+        present = False
+        for line in self.get_show_section("cpu detailed"):
+            x = re.search("^Core\s?\d+",line)
```

Comment:
Be careful with escape sequences in python - what you've done here won't hurt anything and will work, but, if you wanted to match a tab - \t or any other escape sequence it won't work the way you want unless you make it a raw string:

<img width="172" alt="image" src="https://user-images.githubusercontent.com/44329220/99737985-81bce280-2a97-11eb-9f72-c2f7a87a48d1.png">

So it's better to get in the habit of using the r prefix. 

Google's [python class](https://developers.google.com/edu/python/regular-expressions) has a good discussion about this.

Reply:
I went ahead and made the revision after I read the above article. I  can definitely see now how the way I had it before would be prone to cause issues.